src/frontend/shortlist/src/components/layout/signup_login/__tests__/LoginSignup.test.js

```diff
@@ -0,0 +1,317 @@
+/*
+TODO(All these vue files do not have props that correspond to the API, needs to be added)
+The following tests conclude all major component testing.
+*/
+
+import { describe, it, expect, vi } from "vitest";
+import { mount } from "@vue/test-utils";
+import Signup from "../Signup.vue";
+import Login from "../Login.vue";
+import Logo from "../Logo.vue";
+import LoggedOut from "../LoggedOut.vue";
+import ModalFullScreen from "./../../ModalFullScreen.vue";
+// import axios from "axios";
+import { createTestingPinia } from "@pinia/testing";
+
+describe("Login.vue", () => {
+  // Check if login exists
+  it("Renders with null prop", () => {
+    const componentWrapper = mount(Login, {
+      props: {
+        /*
+        TODO(Add API variables here)
+        */
+      },
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    expect(componentWrapper.exists()).toBe(true);
+  });
+
+  it("Check if the login_components_container exists", async () => {
+    const componentWrapper = mount(Login, {
+      props: {},
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    const login_components_container = componentWrapper.find(
+      ".login_components_container"
+    );
+    expect(login_components_container.exists()).toBe(true);
+  });
+
+  it("Check if the loginForm works", async () => {
+    const componentWrapper = mount(Login, {
+      props: {},
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    const loginForm = componentWrapper.find("#loginForm");
+    expect(loginForm.exists()).toBe(true);
+    await loginForm.trigger("submit.prevent");
+  });
+
+  it("Check if the loginThroughSSO works", async () => {
+    const componentWrapper = mount(Login, {
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    const loginWithGoogle = componentWrapper.find("#loginWithGoogle");
+    expect(loginWithGoogle.exists()).toBe(true);
+    await loginWithGoogle.trigger("submit.prevent");
+  });
+
+  it("Check if the switchToSignupButton works", async () => {
+    const componentWrapper = mount(Login, {
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    const switchToSignupButton = componentWrapper.find("#switchToSignupButton");
+    expect(switchToSignupButton.exists()).toBe(true);
+    await switchToSignupButton.trigger("click");
```

Comment:
Same reasoning as above

Reply:
See above. If you think I am missing a lot of similar functions here, this is the exact reason why I needed reviews on day one I created the first few tests